src/native-stack/views/NativeStackView.tsx

```diff
@@ -264,6 +264,7 @@ const RouteView = ({
       key={route.key}
       enabled
       isNativeStack
+      screenDescriptor={descriptors[route.key]}
```

Comment:
Just a small comment from me about this line: I'm creating `screenDescriptor` prop here to access the options for `HeaderConfig` that is not available from the `ScreenProps` prop the `ReanimatedNativeStackScreen` gets, but since we're providing such information - maybe we could provide all of the information to the screen via some sort of hook, like `useScreenProperties`? 

We could provide such information like an info about the route, screen index and descriptors, just as the user is receiving the information about the navigation from `useNavigation` hook. I could also benefit from it by using that hook for receiving an info about the descriptors of the screen.
What do you think?

Reply:
Resolved it in https://github.com/software-mansion/react-native-screens/pull/1915/commits/28a4d92f3bd536b93de1a62beb42e091aa00d49a - as we don't want to push such a big object to the native side, I've replaced it to the hook.